core/dbt/parser/base.py

```diff
@@ -173,7 +209,12 @@ def _update_parsed_node_info(self, parsed_node, config):
         schema_override = config.config.get('schema')
         get_schema = self.get_schema_func()
         parsed_node.schema = get_schema(schema_override).strip()
-        parsed_node.alias = config.config.get('alias', parsed_node.get('name'))
+
+        alias_override = config.config.get('alias')
+        get_alias = self.get_alias_func()
+        temp_node = config.config.get('alias', parsed_node.get('name'))
+        parsed_node.alias = get_alias(temp_node, alias_override).strip()
```

Comment:
What do you think about passing in the actual `parsed_node` here instead of `temp_node` (which is a string). The `parsed_node` will have attributes like
- name
- tags
- package name
- materialization

This would require changes to the `generate_alias_name` macro:
```
{% macro generate_alias_name(node, custom_alias_name=none) -%}

     {%- if custom_alias_name is none -%}

         {{ node.name }}  --<--------- change this line

     {%- else -%}

         {{ custom_alias_name | trim }}

     {%- endif -%}

 {%- endmacro %}
```

By providing a full `node` object, users can encode pretty complex logic when determining an alias for a model! You buy that?

Reply:
Absolutely. Makes sense. To be honest, I'm still getting comfortable with what context is available when with dbt. So I went with the simplest implementation (e.g., passing around strings). I knew you would have a better suggestion.

Reply:
>  To be honest, I'm still getting comfortable with what context is available when with dbt

Us too :). We're working towards normalizing (or at least documenting) all of these contexts, which I think will be a real big help in development!